app/Helpers/Helpers.php

```diff
@@ -123,13 +125,35 @@ function createCacheLock($cache_key, $lock_timeout = 10)
     return Cache::lock($cache_key . '_lock', $lock_timeout);
 }
 
+/**
+ * Set and get the result of callback into cache storage. The key will create from the cache_args array
+ *
+ * @param string $key
+ * @param Array $cache_args
+ * @param Carbon $ttl
+ * @param Closure $callback
+ * @return mixed
+ */
 function cacheRemember($cache_key, $cache_args, $ttl, $callback)
 {
     $key = generateCacheString($cache_key, $cache_args);
+    return cacheRememberByKey($key, $ttl, $callback);
+}
+
+/**
+ * Set and get the result of callback into cache storage usign a given key
+ *
+ * @param string $key
+ * @param Carbon $ttl
+ * @param Closure $callback
+ * @return mixed
+ */
+function cacheRememberByKey(string $key, Carbon $ttl, Closure $callback)
```

Comment:
This new method was created to store the value into cache but the key will be created before. So, we can call the method `generateCacheSafeKey` to create a safe key.